<launch>
  <arg name="CONTROLLER_TYPE" default="auto" />
  <arg name="DEV" default="/dev/input/js0" />
  <arg name="POSE_REMAP" default="joy_pose" />
  <arg name="FRAME_ID" default="base_footprint" />
  <!-- rosparam param="history_pose">[["POSE 1", 1.0,2.1,0.5],['POSE 0',0.0,0.0,0.0],['POSE 2',0.5, 0.3, -1]]</rosparam -->

  <arg name="gui" default="true" />

  <node pkg="joy" type="joy_node" name="joy_driver" output="screen" >
    <param name="dev" type="string" value="$(arg DEV)" />
    <param name="deadzone" value="0.2" />
    <param name="autorepeat_rate" value="40" />
    <param name="coalesce_interval" value="0.025" />
  </node>
  <node pkg="jsk_teleop_joy" type="joy.py" name="joy" output="screen"
        clear_params="true">
    <param name="controller_type" value="$(arg CONTROLLER_TYPE)" />
    <rosparam subst_value="true">
      plugins:
        'JoyTarget6D':
          class: 'JoyTarget6D'
          args:
            publish_pose: True
            pose: $(arg POSE_REMAP)
            frame_id: $(arg FRAME_ID)
            follow_view: true
            triangle_cmd: "PREVIEW"
            circle_cmd: "EXECUTE"
            cross_cmd: "CANCEL"
            save_pose: True
            pose_list: 'history_pose'
            list_length: 5
            save_key: 0
        'SavedPose':
          class: 'DynamicMenu'
          args:
            frame_id: $(arg FRAME_ID)
            topic: 'dynamic_menu'
            title: 'PoseList'
            history: 'history_pose'
            set_pose: 'set_pose'
    </rosparam>
  </node>

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_teleop_joy)/launch/config/dynamic_menu_sample.rviz">
    </node>
  </group>

</launch>
